### ISSUE TITLE
bigip_ltm_virtual_server error out for conditional client_profiles variable

### ISSUE THREAD

**Issue by rsawant23** [F5Networks/terraform-provider-bigip#433]
I am trying to create a virtual server and want to have a client profile on it based on a condition and not have it otherwise. Though Terraform plan says it will create the virtual address with all the correct values. The apply errors out with the following error

```
Error: HTTP 400 :: {"code":400,"message":"01070096:3: Virtual server /Common/vip-name lists profiles incompatible with its protocol.","errorStack":[],"apiError":3}

  on ../../modules/deploy-vip/main.tf line 101, in resource "bigip_ltm_virtual_server" "vip":
 101: resource "bigip_ltm_virtual_server" "vip" {
 ```

### Environment

 * TMOS/Bigip Version: BIG-IP v15.1.0.3 (Build 0.0.12)
 * Terraform Version: v0.13
 * Terraform bigip provider Version: v1.6.0
 
 ###Terraform Code
 
 resource "bigip_ltm_virtual_server" "vip" {
  name                 = "vip-name"
  destination          = "1.1.1.1"
  description          = "test"
  port                   = 80
  pool                 = bigip_ltm_pool.pool.name
  client_profiles      = lower(var.vip-category) == "no-profile" ? []:["vip_profile"]
  source_address_translation = "snat"
  snatpool             = local.source_nat
  irules               = [local.irules]
  profiles             = coalescelist(var.profiles,local.profiles)
  persistence_profiles = var.persistence == "" ? []:[cookie]
  policies             = var.policy_name == "" ? []:[bigip_ltm_policy.policy.name]
}

### Summary

The same logic (conditional) works for the 'persistence' and 'policies' variable but errors out for client_profiles. 
Just a simple client_profiles = [] also gives the same error

### Steps To Reproduce

1. Try to set the client_profile = []

### Expected Behavior
Deploys a vip with no client profile when not required according to the condition

### Actual Behavior
Throws error

**Comment by trinaths** [comment]
created INFRAANO-228 for internal tracking

**Comment by RavinderReddyF5** [comment]
@rsawant23 fixed in v1.8.0
